csrc/scheduler/normalization_utils.cpp

```diff
@@ -635,9 +635,13 @@ getOptionalInnerOuterPersistentBufferBatches(
     const bool ignore_register_size_limit) {
   // if inner_dim_numel <= 1024, we are doing multiple reductions per block
   // with a constant batch size of 1. See Step 5 of
-  // innerOuterPersistentHeuristic
+  // innerOuterPersistentHeuristic. However, if vectorize_factor is 1, we can
+  // increase the persistent batch to 4, which set a minimum workload for each
+  // thread to take advantage of low intra-threads communication cost.
   if (inner_dim_numel <= 1024l) {
-    return std::make_pair(1l, inner_dim_numel / vectorize_factor);
+    const int64_t batch = (vectorize_factor == 1) ? 4l : 1l;
```

Comment:
if vectorize_factor is 1, we can increase the persistent batch to 4, which set a minimum workload for each thread to take advantage of low intra-threads communication cost.

Reply:
Why 4?

Reply:
when vectorized by 2,4,8, batch is 1, this means each thread needs to do serial reduction of 2,4,8 elements. Then, if `vectorize_factor == 1`, we can increase batch size to set a minimum serical reduction workload for each thread. Here I just picked a middle value without spending time to tune as these unvectorized small cases should be rare in real world.

Reply:
That's fine, but please make sure leave it as a code comment

Reply:
yes. added to code comment.